src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1657,10 +1657,19 @@ boolean getAndClearInterrupt() {
         if (com.ibm.oti.vm.VM.isJVMInSingleThreadedMode()) {
             return interruptedImpl();
         }
-        boolean oldValue = interrupted;
+
+        /**
+         * Its possible for the java and native interrupt status to get out of sync. Sending
+         * two interrupts to the same thread in quick succession can cause this. Its critical
+         * that when clearing the interrupt status we also clear the native status. If the
+         * java interrupt status is false, we need to double check that this agrees with the
+         * native status so we dont get an interrupt exception the next time sleep() or wait()
+         * is called.
```

Comment:
```suggestion
         * It is possible for the Java and native interrupt status to get out of sync. Sending
         * two interrupts to the same thread in quick succession can cause this. It is critical
         * that clearing the interrupt status also clears the native status. If the Java
         * interrupt status is false, it is important to verify that this is consistent with the
         * native status to prevent an unexpected interrupt exception the next time
         * sleep() or wait() is called.
```